lib/linguist/heuristics.yml

```diff
@@ -132,6 +132,11 @@ disambiguations:
     pattern: '^\s*(#version|precision|uniform|varying|vec[234])'
   - language: Filterscript
     pattern: '#include|#pragma\s+(rs|version)|__attribute__'
+- extensions: ['.gd']
+  rules:
+  - language: GAP
+    pattern: 'gap> '
```

Comment:
Should all GAP files have this pattern? Our GAP `.gd` sample file doesn't, which is why the Travis build tests are failing.

Reply:
My mistake, the 'gap> ' heuristic is for GAP files with '.tst' extension. Updated to include the correct heuristic for GAP declaration files (.gd extension).

Reply:
Okay. Though my question remains: should all `.gd` GAP files have this `\s*Declare` pattern? Is a `.gd` GAP file without that pattern invalid?

Reply:
So without that statement the program is invalid?

Reply:
That's not technically true, I think it is possible to have a .gd file with no Declare identifiers but it wouldn't do anything interesting (like a source file with only comments). I'm no GAP expert (my interest is actually in GDScript) but this is what I could surmise through code search for .gd files on Github.

Also worth noting: the current state of affairs are flipped, there are ~3x more repositories using [GDScript](https://github.com/search?q=language%3AGDScript) than [GAP](https://github.com/search?q=language%3AGAP), so .gd files misclassifying as GDScript is the more prevalent problem than the other way around.

It is possible to make this heuristic more robust by flipping the pattern and enumerating all valid GDScript identifiers first, but that would be a more inefficient regex. Let me know if you'd like me to explore that option.